.github/workflows/on-push.yml

```diff
@@ -0,0 +1,32 @@
+---
+# Push to main or the stable branches.  The triggering branches also need to
+# be in the triggers at the top of this file.
+name: Push
+on:
+  push:
+    branches: [main, "stable/*"]
+
+jobs:
+  tests-mac:
+    if: github.repository_owner == 'Qiskit'
+    uses: ./.github/workflows/test-mac.yml
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.13"]
+        runner: ["macos-13", "macos-14"]
+    with:
+      python-version: ${{ matrix.python-version }}
+      install-optionals: false
+      runner: ${{ matrix.runner }}
+  test-linux:
+    if: github.repository_owner == 'Qiskit'
+    uses: ./.github/workflows/test-linux.yml
+    strategy:
+      matrix:
+        runner: ["ubuntu-latest", "ubuntu-4.04-arm"]
+    with:
+      python-version: "3.9"
+      test-rust: true
+      test-images: true
+      install-optionals: false
+      runner: ${{ matrix.runner }}
```

Comment:
Do we want to test Windows here too?

Reply:
on azure we only run the linux 3.9 job because it's duplicated effort with the merge queue. The push job is just a sanity check that the merge was fine and doesn't really provide value so we should probably keep it minimal. I'd remove macOS and the arm job from this list too.

Reply:
Removed in ad8cd2e